### PR TITLE
Add sms count to redis

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Add redis cache keys for sms daily counts.

We store the daily notify counts for a service in redis (rather than doing a database query every time!) - here I add in code to put sms counts in redis too.

Note that it's api that actually sets and uses the values, but utils manages the naming 🤷 

Also added an "increment by" function since the sms counts will go up by more than one (for sms with more than one message part)

### CODE REVIEW

File: notifications_utils/clients/redis/redis_client.py [hunk 1]
```diff
@@ -147,6 +147,14 @@ def incr(self, key, raise_exception=False):
             except Exception as e:
                 self.__handle_exception(e, raise_exception, "incr", key)
 
+    def incrby(self, key, by, raise_exception=False):
```
Comment (jzbahrai) [inline notifications_utils/clients/redis/redis_client.py:150]: I dont understand this name but I guess its the same naming convention as above

**Review by jzbahrai — COMMENTED** [review]
lgtm but like umm  this design... why is API doing the work but the naming is happening here 🤔

File: notifications_utils/clients/redis/redis_client.py (hunk 1 above)
Comment (sastels) [reply notifications_utils/clients/redis/redis_client.py:150]: Yeah, basically we're wrapping the redis [incrby](https://redis.io/commands/incrby/) function here, so I kept to the same "use the name of the redis function" that the rest are using.

**Comment by sastels** [comment]
> lgtm but like umm this design... why is API doing the work but the naming is happening here 🤔

I think it's just so that admin can go get the values from redis if it wants to.

**Review by andrewleith — COMMENTED** [review]
Looks good, one question though: what happens when redis restarts or the cache is cleared?  Where is the code that does the initial calculation?

**Comment by sastels** [comment]
> Looks good, one question though: what happens when redis restarts or the cache is cleared? Where is the code that does the initial calculation?

It's handled by api. essentially, when api checks to see if too many have been sent already, if the key isn't in redis it does a database query and sets it. For sms_count it's in this PR:
https://github.com/cds-snc/notification-api/pull/1589/files#diff-22138d151b5f0aba183c2a36804d9ca2cf246fb43f774bb9e1bc3a28cb76aeb7R96-R99

File: notifications_utils/version.py [hunk 2]
```diff
@@ -1,2 +1,2 @@
-__version__ = "48.1.1"
+__version__ = "48.2.0"
```
Comment (andrewleith) [inline notifications_utils/version.py:1]: Version will need to bumped again!

**Review by andrewleith — COMMENTED** [review]
LGTM, just need to up the version number again.

**Comment by jimleroyer** [comment]
About the utils doing the naming discussion: it actually does more. We don't use the actual Redis client directly in admin or API, but rather a class abstraction located in utils that wraps the real client, as showed of the new `incrby` method (which basically proxies and add error management notify style within the app). So utils does a bit more.. but it could improve though to manage better the naming IMO. We can discuss more about it in our code review meetings if we want.

**Review by jzbahrai — APPROVED** [review]
looks good to me other than the version increase. Lets merge :D

**Pull request merged**